### PR TITLE
[DXF export] fixing bugs

### LINKED ISSUE #34841: export dxf improvement
1. If text label contains CR LF signs and these signs are replaced to spaces (qgsdxfexport.cpp, function QgsDxfExport::drawLabel(...), strings 2317-2318). Result text is of one line, but label->getHeight() is height of multiline text. 
So, my idea is to calc count of lines in label than divide label's height by this count.
Something like this:
    int lines_count = 1;
    int start_pos = txt.indexOf(QLatin1String("\r\n"));
    while (start_pos >= 0)
    {
  	  start_pos = txt.indexOf(QLatin1String("\r\n"), start_pos + 1);
  	  lines_count++;
    }
    txt.replace(QLatin1String("\r\n"), QLatin1String(" "));
    start_pos = txt.indexOf('\r');
    while (start_pos >= 0)
    {
  	  start_pos = txt.indexOf('\r', start_pos + 1);
  	  lines_count++;
    }
    txt.replace('\r', ' ');
    start_pos = txt.indexOf('\n');
    while (start_pos >= 0)
    {
  	  start_pos = txt.indexOf('\n', start_pos + 1);
  	  lines_count++;
    }
    txt.replace('\n', ' ');
    writeText(dxfLayer, txt, QgsPoint(label->getX(), label->getY()), label->getHeight() / float(lines_count), label->getAlpha() * 180.0 / M_PI, tmpLyr.format().color());

2. I've got AutoCAD error if layer name contains comma.
Please, add replacement of comma to underscore in function
QString QgsDxfExport::dxfLayerName( const QString &name ):
  layerName.replace( ',', '_');

3. Some AutoCAD versions can't process transparency group code 440.
Please, remove code below  or make it optional in function
void QgsDxfExport::writeGroup( const QColor &color, int exactMatchCode, int rgbCode, int transparencyCode ):
  if ( transparencyCode != -1 && color.alpha() < 255 )
    writeGroup( transparencyCode, 0x2000000 | color.alpha() );

### PR DESCRIPTION
**Pull request by SergeyChern**
## Description
Fixes #34841:
1. Recalc height of multiline labels.
2. Remove commas from layer names.
3. Writing transparency group code 440 for polygon fill color only.

### CODE REVIEW

**Comment by elpaso** [comment]
@SergeyChern please change the PR title to something meaningful.

**Comment by rouault** [comment]
Code layout issues to fix: https://travis-ci.org/github/qgis/QGIS/jobs/659011882?utm_medium=notification&utm_source=github_status

**Comment by SergeyChern** [comment]
> Code layout issues to fix: https://travis-ci.org/github/qgis/QGIS/jobs/659011882?utm_medium=notification&utm_source=github_status

Need help! I've got travis error:
  10/10 Test  #8: qgis_sip_uptodate ................***Failed
  *** SIP file not up to date: core/auto_generated/dxf/qgsdxfexport.sip
I don't understand how to rebuild sip-file.

File: src/core/dxf/qgsdxfexport.cpp [hunk 1]
```diff
@@ -150,7 +150,16 @@ void QgsDxfExport::writeGroup( const QColor &color, int exactMatchCode, int rgbC
 
   int c = ( color.red() & 0xff ) * 0x10000 + ( color.green() & 0xff ) * 0x100 + ( color.blue() & 0xff );
   writeGroup( rgbCode, c );
-  if ( transparencyCode != -1 && color.alpha() < 255 )
+
+  if ( transparencyCode > 0 )
+    return;
```
Comment (m-kuhn) [inline src/core/dxf/qgsdxfexport.cpp]: Does this check do anything? It's the end of the method so the return is ahead anyway

Comment (SergeyChern) [reply src/core/dxf/qgsdxfexport.cpp]: > Does this check do anything? It's the end of the method so the return is ahead anyway

You are right, thanks. I deleted these rows. BTW, can you help me to resolve this error:
*** SIP file not up to date: core/auto_generated/dxf/qgsdxfexport.sip

Comment (m-kuhn) [reply src/core/dxf/qgsdxfexport.cpp]: You need to rebase or merge master into your branch.

**Comment by m-kuhn** [comment]
Are you able to run `./scripts/sipify-all.sh` on your branch?

**Pull request closed without merging**